### PR TITLE
feat(storage): prove native filesystem publication semantics

### LINKED ISSUE #779: feat(storage): prove native filesystem publication semantics
## Parent

Canonical close gate: #776.

## Problem

GraphForge does not have one reusable Rust implementation that proves the native locking, atomic namespace publication, file-flush, platform namespace-durability, and locked-identity semantics required by ADR 0013 on Linux, macOS, and supported Windows NTFS volumes. The implementation must not relax the unsafe-code policy of graphforge-storage.

## Objective

Provide the bounded native filesystem probe backend consumed by project admission, with explicit platform behavior and stable typed outcomes.

## Requirements

- Implement native Linux, macOS, and Windows NTFS probe backends behind one Rust-owned interface.
- Preserve graphforge-storage unsafe-code policy by isolating platform FFI in a narrowly scoped published first-party crate if required.
- Prove exclusive locking, same-filesystem atomic creation/replacement, file flush, stable locked-file identity, and the platform-native namespace durability barrier.
- On POSIX, use the documented file and directory `fsync` contract.
- On Windows NTFS, create/open the staging file with `FILE_FLAG_WRITE_THROUGH`, flush file contents, and rename through the same handle with `SetFileInformationByHandle` / `FILE_RENAME_INFO`; use `ReplaceIfExists = FALSE` for no-replace and the documented replace form for replacement.
- Do not use `ReplaceFileW` or `FlushFileBuffers` on a directory handle as Windows durability proof. Reconcile source/target identity and names after every namespace-operation error before returning.
- Reject ReFS as unproven, along with links, hard-link substitution, traversal, special files, cross-device replacement, network/removable volumes, and unknown filesystem classes.
- Bound artifact count, names, bytes, cleanup, and emitted metadata.
- Return stable typed outcomes suitable for `GF_UNSUPPORTED_FILESYSTEM` mapping.
- Document the trusted-storage assumption: GraphForge cannot prove durability when hardware falsely acknowledges write-through completion.

## Acceptance Criteria

- [ ] Native probe backends pass required Linux, macOS, and Windows NTFS CI.
- [ ] Windows ReFS and all other unsupported/unproven classes return the stable rejection before project mutation.
- [ ] Failure injection covers lock, write, file flush, replacement, namespace durability, identity revalidation, and cleanup.
- [ ] Windows tests cover write-through handle creation, atomic no-replace, atomic replacement, competing destinations, and ambiguous-error reconciliation.
- [ ] Probe artifacts are private, bounded, content-free, and cannot become project authority.
- [ ] Cargo, Bazel, release, license, and notice inventories include any new published first-party crate.
- [ ] graphforge-storage retains its existing unsafe-code prohibition.

## BDD Completion Scenarios

- **Given** a supported local filesystem, **when** the native probe completes, **then** every required publication primitive is proven and all bounded artifacts are removed.
- **Given** Windows NTFS on write-through-capable storage, **when** GraphForge flushes a staging file and renames it through its write-through handle, **then** the probe treats the documented NTFS data-and-rename metadata barrier as complete.
- **Given** Windows ReFS or another unsupported/unproven filesystem, **when** the probe runs, **then** it returns a stable typed rejection without touching a project root.
- **Given** a Windows namespace operation reports failure, **when** GraphForge reconciles source and target identities and names, **then** it reports the observed outcome and leaves no false success or unbounded artifact.

## Non-Goals

Project-root creation coordination, lifecycle call-site integration, and public publisher routing; those belong to sibling #780. ReFS durability and guarantees for storage hardware that dishonestly acknowledges write-through are explicitly out of scope.

### PR DESCRIPTION
**Pull request by DecisionNerd**
## Summary

- add the published `graphforge-filesystem` crate for audited handle-scoped publication primitives while preserving `graphforge-storage`'s unsafe-code prohibition
- provide bounded native filesystem admission probes with cross-process locking, stable identity, typed error reconciliation, and fail-closed link/reparse handling
- amend the Windows durability contract through ADR 0020: support fixed writable local NTFS on honest write-through storage, rename through a flushed `FILE_FLAG_WRITE_THROUGH` source handle with `SetFileInformationByHandle`, and reject ReFS as unproven
- remove `ReplaceFileW` and directory `FlushFileBuffers` as Windows durability authority while retaining atomic no-replace/replacement behavior and source/target reconciliation
- lock the exact POSIX/NTFS filesystem scope and platform-native namespace barrier into the durability matrix and repository policy gates
- harden workflow-policy parsing so comments, nested YAML, unrelated text, background execution, failure suppression, and shell control-flow bypasses cannot impersonate required CI commands
- keep required macOS and Windows native jobs in CI Gate and preserve the complete Cargo/Bazel/release/license/notice inventory

## Scenario evidence

- supported host probe: 4/4 native primitive tests and 9/9 storage admission tests passed
- lock and crash behavior: subprocess contention and crash-release tests passed
- failure phases: typed write, flush, replace, unknown-state, identity, namespace-durability, and cleanup tests passed with bounded cleanup
- Windows compile: `graphforge-filesystem` tests compile for `x86_64-pc-windows-msvc`; native execution is required from exact-head hosted Windows CI
- contract: ADR 0020 explicitly amends ADR 0013/0018; the exact supported/rejected filesystem set and M5 interchange vocabulary are validated by the machine-readable durability gate
- CI policy: adversarial fixtures reject inactive YAML text and shell failure-suppression/control-flow forms while retaining ordinary redirections

## Local verification

- `cargo fmt --all -- --check`
- `cargo clippy -p graphforge-filesystem --all-targets -- -D warnings`
- `cargo clippy -p graphforge-storage --lib -- -D warnings`
- `cargo test -p graphforge-filesystem --lib --no-fail-fast` — 4 passed
- `cargo test -p graphforge-storage filesystem_admission::tests:: --lib --no-fail-fast` — 9 passed
- `cargo check -p graphforge-filesystem --tests --target x86_64-pc-windows-msvc`
- `python3 scripts/ci/test-binding-release-candidate.py`
- `python3 scripts/ci/test-ci-storage-policy.py`
- `python3 scripts/ci/durability-isolation-gate.py validate`
- `python3 scripts/ci/test-durability-isolation-gate.py`
- Cargo/Bazel parity, drift, ledger, release-policy, notice, license, package, and publish-dry-run checks from the prior exact-head run

Closes #779

## Summary by CodeRabbit

* **New Features**
  * Added cross-platform filesystem durability checks before project storage is used.
  * Validates permissions, supported volumes, file safety, locking, atomic replacement, and cleanup.
  * Added reliable filesystem identity, link-count, private-directory, and safe file-installation capabilities.
  * Added clearer diagnostics for unsupported or unsafe filesystem configurations.

* **Bug Fixes**
  * Improved handling and reporting of uncertain file-replacement outcomes.

* **Tests**
  * Expanded coverage for Unix and Windows behavior, concurrency, crash recovery, and filesystem edge cases.

### CODE REVIEW

File: crates/graphforge-storage/src/filesystem_admission.rs [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+        _ => return Err(unsupported("CLASSIFY", "filesystem_class_unproven")),
+    };
+    let vfs = rustix::fs::statvfs(parent)
+        .map_err(|_| unsupported("CLASSIFY", "native_volume_query_failed"))?;
+    if vfs.f_flag.contains(rustix::fs::StatVfsMountFlags::RDONLY) {
+        return Err(unsupported("CLASSIFY", "volume_read_only"));
+    }
+    reject_removable_volume(parent)?;
+    Ok(class.into())
+}
```
[the remark this answers is not in the document]
Comment (DecisionNerd) [reply crates/graphforge-storage/src/filesystem_admission.rs:292]: Verified against the authoritative hosted Bazel lane at PR head 6a473201: //:ci_rust_tests passed on the configured Linux runner. Skipping unsupported filesystem classes would weaken this fail-closed durability test and violates the repository no-skip policy. No change is appropriate; the allowlist and assertions remain strict.

File: scripts/ci/workflow_policy.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            if current is not None:
+                jobs[current] = "\n".join(body)
+            current = match.group(1)
+            body = []
+            continue
+        if current is None:
+            continue
+        if line and not line.startswith("  "):
+            break
+        body.append(line)
```
[the remark this answers is not in the document]
Comment (DecisionNerd) [reply scripts/ci/workflow_policy.py:39]: Fixed at e8f6189. The parser now ignores top-level comment separators without truncating later jobs and requires each required run scalar to start with the command. Adversarial fixtures cover conditional/loop wrappers, uninvoked functions, heredocs, explicit , and existing shell-suppression forms. The two real workflows with safe  preludes were split so their required scalars remain directly executable and fail closed.

File: tests/unit/test_set_release_version.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from pathlib import Path
 
 import pytest
+import tomllib
```
[the remark this answers is not in the document]
Comment (DecisionNerd) [reply tests/unit/test_set_release_version.py]: Fixed at e8f6189. The test now imports  with a  fallback for the declared Python 3.10 floor, and derives the lock-package count from the manifest package set instead of duplicating . The focused release-version test passes.
